platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -538,6 +538,10 @@ + (instancetype)mgl_expressionWithJSONObject:(id)object {
                 return [NSExpression expressionForAggregate:MGLSubexpressionsWithJSONObjects(argumentObjects.firstObject)];
             }
             return [NSExpression mgl_expressionWithJSONObject:argumentObjects.firstObject];
+        } else if ([op isEqualToString:@"number"] ||
+                   [op isEqualToString:@"string"] ||
+                   [op isEqualToString:@"boolean"]) {
+            return [NSExpression mgl_expressionWithJSONObject:argumentObjects.firstObject];
```

Comment:
I’m not sure it’s a good idea to make the conversion from JSON to NSExpression lossy like this. If these calls to `number` etc. are failing the round-tripping tests, feel free to modify the tests to start out with these calls so that they pass without needing to remove the type assertions.